### ISSUE TITLE
Process leak when unable to find exec-credential binary

### ISSUE THREAD

**Issue by ragarcia26** [kubernetes/kubectl#1492]
**What happened**:
We use custom tooling (a krew installed kube plugin) for users to authenticate with kube clusters using the [exec credential auth flow](https://kubernetes.io/docs/reference/access-authn-authz/authentication/#client-go-credential-plugins). Similar to that of [kubelogin](https://github.com/int128/kubelogin). We've found that if a client using the exec credential config is unable to find the binary, it'll start to spawn many kubectl processes bogging down cpu/memory. For example, this can occur due to ./krew not being in the path. This was discovered when someone using the vscode plugin [Google Cloud Code ](https://cloud.google.com/code/docs/vscode/install) on windows noticed vscode wasn't reading the %USERPROFILE% var correctly in the PATH and only after manually adding the full .krew path to the PATH did it start to work. We have had reports that this has also occured with K9's. So while we now understand the issue, potential issue conditions, and the workaround, it seems odd that kubectl will start infinitely process spawning. 

**What you expected to happen**:
Kubectl does not spawn new processes when it can't use the exec credentials auth and shortly returns an error.

**How to reproduce it (as minimally and precisely as possible)**:

1. Have a cluster context that uses a kubectl plugin installed via krew to auth via [exec credentials ](https://kubernetes.io/docs/reference/access-authn-authz/authentication/#client-go-credential-plugins)
2. Open a process monitor (activity monitor for mac, task manager for windows) to see kubectl processes running
3. Either move/rename the plugin or remove the .krew path from the PATH
4. run kubectl get po -A 

**Anything else we need to know?**:

**Environment**:
***Simple Testing*** 
- Kubernetes client and server versions (use `kubectl version`):
```
WARNING: This version information is deprecated and will be replaced with the output from kubectl version --short.  Use --output=yaml|json to get the full version.
Client Version: version.Info{Major:"1", Minor:"27", GitVersion:"v1.27.2", GitCommit:"7f6f68fdabc4df88cfea2dcf9a19b2b830f1e647", GitTreeState:"clean", BuildDate:"2023-05-17T14:13:27Z", GoVersion:"go1.20.4", Compiler:"gc", Platform:"darwin/arm64"}
Kustomize Version: v5.0.1
Server Version: version.Info{Major:"1", Minor:"25+", GitVersion:"v1.25.12-eks-2d98532", GitCommit:"0aa16cf4fac4da27b9e9e9ba570b990867f6a3d8", GitTreeState:"clean", BuildDate:"2023-07-28T16:52:04Z", GoVersion:"go1.20.6", Compiler:"gc", Platform:"linux/amd64"}
WARNING: version difference between client (1.27) and server (1.25) exceeds the supported minor version skew of +/-1
```
- Cloud provider or hardware configuration:
aws
- OS (e.g: `cat /etc/os-release`):
monterey
ProductName:    macOS
ProductVersion: 12.6.9
BuildVersion:   21G726

***VSCode User Testing***
- Kubernetes client and server versions (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"23+", GitVersion:"v1.23.14-dispatcher-dirty", GitCommit:"35498c8a8d141664928467fda116cd500d09bc21", GitTreeState:"dirty", BuildDate:"2022-11-16T20:14:00Z", GoVersion:"go1.17.13", Compiler:"gc", Platform:"windows/amd64"}
```
- Cloud provider or hardware configuration:
aws
- OS (e.g: `cat /etc/os-release`):
windows 10

**Comment by ardaguclu** [comment]
client-go executes plugin in https://github.com/kubernetes/kubernetes/blob/f638e2849fbddeb201b8290586a4747af99086bf/staging/src/k8s.io/client-go/plugin/pkg/client/auth/exec/exec.go#L436 by using the builtin function in go. I doubt that we have an option to fix this problem inside of k8s codebase.

**Comment by ardaguclu** [comment]
I modified the kubeconfig to use non-existed cred exec plugin to reproduce this issue. After executing the kubectl, I saw cred exec plugin execution error. In another terminal, when I checked the processes via `ps -aux`, I couldn't see spawned processes. 

Is it possible that some other process(for instance, vscode) causes this process spawning because there is no reason that kubectl can cause this.

/triage needs-information
